qutebrowser/browser/webkit/webkithistory.py

```diff
@@ -48,7 +50,8 @@ def historyContains(self, url_string):
         Return:
             True if the url is in the history, False otherwise.
         """
-        return url_string in self._history.history_dict
+        with debug.log_time('sql', 'historyContains'):
+            return url_string in self._history
```

Comment:
Hmm, you're right... I don't like the idea of the simple infinite set though, due to #1476.